app/src/main/java/com/keylesspalace/tusky/adapter/StatusBaseViewHolder.java

```diff
@@ -1069,7 +1071,16 @@ protected void setupCard(StatusViewData.Concrete status, CardViewMode cardViewMo
                 cardImage.setImageResource(R.drawable.card_image_placeholder);
             }
 
-            cardView.setOnClickListener(v -> LinkHelper.openLink(card.getUrl(), v.getContext()));
+            // View embedded photos in our image viewer instead of opening the browser
+            if (card.getType().equals(CardKt.TYPE_PHOTO) && !TextUtils.isEmpty(card.getEmbed_url())) {
+                cardImage.setOnClickListener(v ->  {
+                    cardView.getContext().startActivity(ViewMediaActivity.newSingleImageIntent(cardView.getContext(), card.getEmbed_url()));
+                });
+                cardInfo.setOnClickListener(v -> LinkHelper.openLink(card.getUrl(), v.getContext()));
+            } else {
+                cardView.setOnClickListener(v -> LinkHelper.openLink(card.getUrl(), v.getContext()));
+            }
```

Comment:
For view recycling reasons you should set the click listener that is not used to `null` in the respective branch otherwise it might open unexpected things